src/com/amazon/ion/impl/lite/IonValueLite.java

```diff
@@ -979,10 +990,21 @@ protected int hashTypeAnnotations(final int original)
             return original;
         }
 
-        int result = valueHashSalt * original + tokens.length;
+        // Note: Because the tokens array doubles in size when it grows, tokens.length does not correctly convey
+        // the actual number of annotations on the value.
+        int count = 0;
+        for (SymbolToken token : tokens) {
+            if (token == null) {
+                break;
+            }
+            count++;
+        }
+
+        int result = valueHashSalt * original + count;
```

Comment:
This is an unrelated fix, now covered by `IonValueTest.hashValueWithEmptySpaceInItsAnnotationsArray`.